src/Centreon/Domain/RemoteServer/RemoteServerService.php

```diff
@@ -0,0 +1,153 @@
+use Centreon\Domain\PlatformTopology\Platform;
+use Centreon\Domain\PlatformTopology\PlatformException;
+use Centreon\Domain\RemoteServer\RemoteServerException;
+use Centreon\Domain\Menu\Interfaces\MenuRepositoryInterface;
+use Centreon\Domain\RemoteServer\Interfaces\RemoteServerServiceInterface;
+use Centreon\Domain\PlatformTopology\Interfaces\PlatformTopologyRepositoryInterface;
+use Centreon\Domain\RemoteServer\Interfaces\RemoteServerRepositoryInterface;
+
+class RemoteServerService implements RemoteServerServiceInterface
+{
+
+    /**
+     * @var MenuRepositoryInterface
+     */
+    private $menuRepository;
+
+    /**
+     * @var PlatformTopologyRepositoryInterface
+     */
+    private $platformTopologyRepository;
+
+    /**
+     * @var RemoteServerRepositoryInterface
+     */
+    private $remoteServerRepository;
+
+    /**
+     * @var string
+     */
+    private $centreonEtcPath;
+
+    /**
+     * @param MenuRepositoryInterface $menuRepository
+     * @param PlatformTopologyRepositoryInterface $platformTopologyRepository
+     */
+    public function __construct(
+        MenuRepositoryInterface $menuRepository,
+        PlatformTopologyRepositoryInterface $platformTopologyRepository,
+        RemoteServerRepositoryInterface $remoteServerRepository
+    ) {
+        $this->menuRepository = $menuRepository;
+        $this->platformTopologyRepository = $platformTopologyRepository;
+        $this->remoteServerRepository = $remoteServerRepository;
+    }
+
+    public function setCentreonEtcPath(string $centreonEtcPath): void
+    {
+        $this->centreonEtcPath = DIRECTORY_SEPARATOR . ltrim($centreonEtcPath, DIRECTORY_SEPARATOR);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function convertCentralToRemote(): void
+    {
+        /**
+         * Stop conversion if the Central has remote children
+         */
+        try {
+            $platformChildren = $this->platformTopologyRepository->findCentralRemoteChildren();
+            if (!empty($platformChildren)) {
+                throw new RemoteServerException(
+                    _("Your Central is linked to another remote(s), conversion in Remote isn't allowed")
+                );
+            }
+        } catch (RemoteServerException $ex) {
+            throw $ex;
+        } catch (\Exception $ex) {
+            throw new RemoteServerException(_('An error occured while searching any remote children'));
+        }
+
+        /**
+         * Set Remote type into Platform_Topology
+         */
+        $this->updatePlatformTypeParameters(Platform::TYPE_REMOTE);
+
+        try {
+            $this->menuRepository->disableCentralMenus();
+        } catch (\Exception $ex) {
+            throw new MenuException(_('An error occured while disabling the central menus'));
+        }
+
+        /**
+         * Apply Remote Server mode in configuration file
+         */
+        $this->remoteServerRepository->updateInstanceMod($this->centreonEtcPath . "conf.pm", "remote");
```

Comment:
A repository must remain abstract in its use without revealing the technique used to process the information.
Here you pass a filename, please to inject it into the repository instead.